src/main/java/soot/SootMethodRefImpl.java

```diff
@@ -360,12 +328,12 @@ public String toString() {
   public int hashCode() {
     final int prime = 31;
     int result = 1;
-    NumberedString subsig = getSubSignature();
     result = prime * result + ((declaringClass == null) ? 0 : declaringClass.hashCode());
     result = prime * result + (isStatic ? 1231 : 1237);
     result = prime * result + ((name == null) ? 0 : name.hashCode());
     result = prime * result + ((parameterTypes == null) ? 0 : parameterTypes.hashCode());
     result = prime * result + ((returnType == null) ? 0 : returnType.hashCode());
+    NumberedString subsig = getSubSignature();
     result = prime * result + ((subsig == null) ? 0 : subsig.hashCode());
```

Comment:
in general: isn't it redundant to add the SubSignature to the hash and its parts (name,parameters, returntype) as well?

Reply:
I agree, it does seem redundant. Should we go ahead and remove it from the hashcode?

Reply:
yes i think so (and equals if applicable)

Reply:
I removed the subsig sections from both of those methods.